grakn-engine/src/main/java/ai/grakn/engine/controller/GraqlController.java

```diff
@@ -102,16 +102,19 @@ private Json executeGraql(Request request, Response response){
         String queryString = mandatoryQueryParameter(request, QUERY);
         boolean infer = parseBoolean(mandatoryQueryParameter(request, INFER));
         boolean materialise = parseBoolean(mandatoryQueryParameter(request, MATERIALISE));
+        Json responseBody;
 
-        try(GraknGraph graph = factory.getGraph(keyspace, READ)){
+        try(GraknGraph graph = factory.getGraph(keyspace, WRITE)){
             Query<?> query = graph.graql().materialise(materialise).infer(infer).parse(queryString);
 
             if(!readOnly(query)){
                 throw new GraknEngineServerException(405, "Only \"read-only\" queries are allowed.");
             }
-
-            return respond(request, query, response);
+            //Do not return directly inside the try block otherwise open transactions will not autocommit and close properly.
+            responseBody = respond(request, query, response);
```

Comment:
Ah, that explains it, I had forgotten about that part. Since I didn't expect such a prompt response from Filipe, I already merged the PR, so we should make (and test) a fix with the commit call.